File: application/tests/model_selector_test.rs
#[test]
fn model_selector_defaults() {
    let haiku = "anthropic.claude-haiku-4-5-20251001-v1:0";
    let sonnet = "anthropic.claude-sonnet-4-5-20250929-v1:0";
    let opus = "anthropic.claude-opus-4-6-v1";

    assert!(haiku.contains("haiku"));
    assert!(sonnet.contains("sonnet"));
    assert!(opus.contains("opus"));
}

#[test]
fn model_ids_are_valid_format() {
    let models = [
        "anthropic.claude-haiku-4-5-20251001-v1:0",
        "anthropic.claude-sonnet-4-5-20250929-v1:0",
        "anthropic.claude-opus-4-6-v1",
    ];

    for model in models {
        assert!(model.starts_with("anthropic.claude-"));
        assert!(model.contains("claude"));
    }
}

File: application/tests/model-selector.test.ts
import { describe, it, expect } from 'vitest';
import { selectModel } from '../src/llm/model-selector.js';

describe('selectModel', () => {
  it('returns haiku for simple greetings', () => {
    expect(selectModel([{ role: 'user', content: 'hello' }])).toBe('bedrock.claude-haiku-4-5');
    expect(selectModel([{ role: 'user', content: 'hi' }])).toBe('bedrock.claude-haiku-4-5');
    expect(selectModel([{ role: 'user', content: 'thanks' }])).toBe('bedrock.claude-haiku-4-5');
  });

  it('returns sonnet for standard queries', () => {
    expect(selectModel([{ role: 'user', content: 'How do I use async/await in TypeScript?' }]))
      .toBe('bedrock.claude-sonnet-4-6');
  });

  it('returns opus for complex reasoning tasks', () => {
    expect(selectModel([{ role: 'user', content: 'Analyze the trade-offs between microservices and monolith architecture for our e-commerce platform' }]))
      .toBe('bedrock.claude-opus-4-6');
  });

  it('returns opus for long inputs (>200 chars)', () => {
    const longInput = 'a'.repeat(201);
    expect(selectModel([{ role: 'user', content: longInput }]))
      .toBe('bedrock.claude-opus-4-6');
  });

  it('returns sonnet as default when no messages', () => {
    expect(selectModel([])).toBe('bedrock.claude-sonnet-4-6');
  });

  it('handles multimodal content parts', () => {
    expect(selectModel([{
      role: 'user',
      content: [{ type: 'text', text: 'hello' }],
    }])).toBe('bedrock.claude-haiku-4-5');
  });
});
